blog/en/blog/2023/08/11/rate-limit.md

```diff
@@ -0,0 +1,282 @@
+
+You can always set this up in your services directly without configuring it in APISIX. But as the number of your services increases, with each service having different constraints, setting up and managing different rate limits and updating them might become difficult.
+
+In this article, we will look at examples of how we can use the rate limiting plugins in APISIX. You can find the complete configuration files and instructions to deploy for this article in [this repository](https://github.com/navendu-pottekkat/rate-limit).
+
+## Rate Limit All Requests
+
+Setting up a rate limit for all your requests is a good place to start.
+
+APISIX allows configuring [global rules](https://apisix.apache.org/docs/apisix/terminology/global-rule/) that are applied on all requests to APISIX. We can configure a global rule with the `limit-req` plugin, as shown below:
+
+```shell
+curl localhost:9180/apisix/admin/global_rules/rate_limit_all -X PUT -d '
+{
+  "plugins": {
+    "limit-req": {
+      "rate": 30,
+      "burst": 30,
+      "key_type": "var",
+      "key": "remote_addr",
+      "rejected_code": 429,
+      "rejected_msg": "rate limit exceeded!"
+    }
+  }
+}'
+```
+
+The configuration above will allow 30 requests per second (rate) to your upstream services. When the number of requests exceeds 30 but is under 60 (burst), the additional requests are delayed to match the 30 requests per second limit. If the number of requests exceeds 60, the additional requests are rejected. In short, the upstream services will always have a maximum of 30 requests per second.
+
+A global rate limit can be helpful when you want to set hard limits on the number of requests your upstream should receive in a second. But sometimes, you might want to rate limit the requests to a particular route instead of all requests.
+
+To do this, you can configure the plugin on a specific route. The example below shows how you can configure the `limit-count` plugin on a particular route:
+
+```shell
+curl localhost:9180/apisix/admin/routes/rate_limit_route -X PUT -d '
+{
+  "uri": "/api",
+  "upstream": {
+    "type": "roundrobin",
+    "nodes": {
+      "upstream:80": 1
+    }
+  },
+  "plugins": {
+    "limit-count": {
+      "count": 10,
+      "time_window": 60,
+      "key_type": "var",
+      "key": "remote_addr",
+      "policy": "local",
+      "rejected_code": 429,
+      "rejected_msg": "rate limit exceeded!"
+    }
+  }
+}'
+```
+
+This will allow only ten requests in a minute, and this will be reset every minute.
+
+A global and route level limit combo can be used to set hard limits on top of use case-specific limits, which is particularly helpful in many practical scenarios.
```

Comment:
Please confirm how they play together, _i.e._, that the lowest threshold hit is the limit